traits/tests/test_traits_listener.py

```diff
@@ -0,0 +1,351 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+"""
+Unittest tests on ListenerParser, ListenBase etc.
+See test_listeners for integration tests on the public API using these objects.
+
+"""
+
+import unittest
+
+from traits import traits_listener
+from traits.api import (
+    TraitError,
+    pop_exception_handler,
+    push_exception_handler,
+)
+
+
+class TestListenerParser(unittest.TestCase):
+
+    def setUp(self):
+        push_exception_handler(
+            handler=lambda *args: None, reraise_exceptions=True)
+
+    def tearDown(self):
+        pop_exception_handler()
+
+    def test_listener_parser_single_string(self):
+        text = "some_trait_name"
+        parser = traits_listener.ListenerParser(text=text)
+
+        listener = parser.listener
+        self.assertEqual(listener.name, "some_trait_name")
```

Comment:
Would it make sense to define `__eq__` on `ListenerItem` and related types, just  to make this sort of test easier?

Reply:
I did not want to add `__eq__` (and its friends) just for the tests, so I added `addTypeEqualityFunc` instead. In the situation when the `ListenerItem.next` is a `ListenerGroup` and that we need to check `ListenerGroup.items`, we still need to loop over them.  I did not go to the length of defining another equality function for `ListenerGroup`.

Reply:
Sounds sane to me.